x/concentrated-liquidity/lp.go

```diff
@@ -384,3 +408,29 @@ func emitLiquidityChangeEvent(ctx sdk.Context, eventType string, positionId uint
 		sdk.NewAttribute(types.AttributeAmount1, actualAmount1.String()),
 	))
 }
+
+// isLockMature checks if the underlying lock has expired.
+// If the lock doesn't exist, it returns true.
+// If the lock exists, it checks if the lock has expired.
+// If the lock has expired, it returns true.
+// If the lock is still active, it returns false.
+func (k Keeper) isLockMature(ctx sdk.Context, underlyingLockId uint64) (bool, error) {
+	// Query the underlying lock
+	underlyingLock, err := k.lockupKeeper.GetLockByID(ctx, underlyingLockId)
+	if err != nil && errors.Is(err, lockuptypes.ErrLockupNotFound) {
+		// Lock doesn't exist, so we can withdraw from this position
+		return true, nil
+	} else if err != nil {
+		// Unexpected error, return false to prevent any further action and return the error
+		return false, err
+	}
+
+	// Check if the lock has expired
+	if underlyingLock.EndTime.After(ctx.BlockTime()) {
+		// Lock is still active, so we cannot withdraw from this position
+		return false, nil
+	}
+
+	// Lock has expired, so we can withdraw from this position
+	return true, nil
+}
```

Comment:
isLockMature test gets added in the next PR here https://github.com/osmosis-labs/osmosis/pull/4883/files#diff-8a43021241fc488af68dd6cdd7c762a85476f3aa8e44248784e9265c9ef0bbdf

This is because the isLockMature test relies on createFullRangePosition unlocking method that is not introduced until the next PR